webapp/crashstats/crashstats/templatetags/jinja_helpers.py

```diff
@@ -157,21 +147,17 @@ def show_bug_link(bug_id):
     return markupsafe.Markup(tmpl) % data
 
 
-EXTRA_NEWLINES_RE = re.compile(r"\n\n\n+")
-
 MAX_TITLE_LENGTH = 255
 
 
 @library.global_function
-def generate_create_bug_url(
-    request, template, raw_crash, report, parsed_dump, crashing_thread
-):
-    parsed_dump = parsed_dump or {}
-
+def generate_create_bug_url(crash_report_url, template, processed_crash):
```

Comment:
I made big changes to the signature here:

I don't want to pass the request into a utility function because that requires us to create a lot of HTTP state which we shouldn't be requiring for a utility function. By dropping the request, it's possible for someone else (like Suhaib) to copy our link generation code verbatim without having to make changes to it.

However, we need the full url to the crash report which depends on service details that are in the request--the scheme and host. Thus, we now pass in the fully-formed crash report url and the caller can generate that and the utility function doesn't have to know anything about requests.

This function and downstream functions don't use the raw crash data, so I removed that from the arguments.

The parsed dump is really the `processed_crash['json_dump']` with a bunch of processing that gets used on the Details page of the crash report, but no where else--including in `generate_create_bug_url`, so we can use `processed_crash['json_dump']` and drop `parsed_dump` from the arguments.